torch/csrc/Dtype.cpp

```diff
@@ -25,6 +28,19 @@ PyObject* THPDtype_New(at::ScalarType scalar_type, const std::string& name) {
   END_HANDLE_TH_ERRORS
 }
 
+PyObject* THPDtype_pynew(PyObject* type, PyObject* args, PyObject* kwargs) {
```

Comment:
nit: Maybe rename to something like THPDtype_pyNewCustomDtype 
THPDtype_pynew reads like something that goes in tp_new

A small comment mentioning the limitations of this API would be good.